[Zephyr] Build error due to missing `asm_inline.h` header

**MCUboot commit:** db974f2fc23e4cfeb423b0665f1907273fdbcdee
**Zephyr commit:** 9afb0cfab010519d92c4706e5070b37e8725505a

Attempting to build MCUboot results in the following build error:
```
/Users/ccollins/repos/mcuboot/boot/zephyr/main.c:22:24: fatal error: asm_inline.h: No such file or directory
```

Applying the following patch fixes the build error:
```
diff --git a/boot/zephyr/main.c b/boot/zephyr/main.c
index 1159e34..1f80dac 100644
--- a/boot/zephyr/main.c
+++ b/boot/zephyr/main.c
@@ -19,7 +19,6 @@
 #include <gpio.h>
 #include <misc/__assert.h>
 #include <flash.h>
-#include <asm_inline.h>
 #include <drivers/system_timer.h>

 #include "target.h"
@@ -65,7 +64,7 @@ static void do_boot(struct boot_rsp *rsp)
                                      rsp->br_hdr->ih_hdr_size);
     irq_lock();
     sys_clock_disable();
-    _MspSet(vt->msp);
+    __set_MSP(vt->msp);
     ((void (*)(void))vt->reset)();
 }
 #else
```

But this may introduce build failures with older versions of Zephyr.

The Zephyr commit that causes this problem is:
```
commit c028f88b37d2a033d6a85c8d365ef048e1e4d621
Author: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>
Date:   Mon Mar 12 10:45:02 2018 +0100

    arch: arm: remove redundant asm inline headers

    This commit removes the unnecessary asm inline header for ARM.
    It also adapts the stack.h and exc.h to use the ARM CMSIS inline
    functions to access the IPSR and MSP registers.

    Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>
```

--- a/boot/zephyr/main.c
+++ b/boot/zephyr/main.c
@@ -19,7 +19,6 @@
 #include <gpio.h>
 #include <misc/__assert.h>
 #include <flash.h>
-#include <asm_inline.h>
 #include <drivers/system_timer.h>
 
 #include "target.h"
@@ -65,7 +64,7 @@ static void do_boot(struct boot_rsp *rsp)
                                      rsp->br_hdr->ih_hdr_size);
     irq_lock();
     sys_clock_disable();
-    _MspSet(vt->msp);
+    __set_MSP(vt->msp);
     ((void (*)(void))vt->reset)();
 }
 #else

